custom/enikshay/integrations/bets/views.py

```diff
@@ -99,56 +118,69 @@ def get_case(domain, case_id):
     return case_accessor.get_case(case_id)
 
 
-def _get_case_update(request, domain, update_model):
+def _get_case_updates(request, domain):
     try:
         request_json = json.loads(request.body)
     except ValueError:
         raise ApiError(msg="Malformed JSON", status_code=400)
-    try:
-        update = update_model.wrap(request_json)
-    except BadValueError as e:
-        raise ApiError(msg=e.message, status_code=400)
-    try:
-        case = get_case(domain, update.case_id)
-        if case.type != update.case_type:
-            raise CaseNotFound()
-    except CaseNotFound:
+
+    if not isinstance(request_json.get('response', None), list):
+        raise ApiError(msg='Expected json of the form `{"response": []}`', status_code=400)
+
+    updates = []
+    for event_json in request_json['response']:
+        if event_json.get('event_type', None) not in ('Voucher', 'Incentive'):
+            raise ApiError(msg="Malformed JSON", status_code=400)
```

Comment:
If this is ever going to happen, would be nice for the error to be more specific.